src/datadog/trace_segment.cpp

```diff
@@ -245,6 +250,31 @@ void TraceSegment::span_finished() {
     span.tags[tags::internal::runtime_id] = runtime_id_.string();
   }
 
+  // calculate http.endpoint if:
+  // a) the feature is not disabled, and
+  // b) the tag http.endpoint is not already set, and
+  // c) http.url is set, and
+  // d) http.route is not set or resource_renaming_mode is ALWAYS_CALCULATE
+  if (resource_renaming_mode_ != HttpEndpointCalculationMode::DISABLED &&
```

Comment:
I agree with Damien's original comment that this operation can be refactored into its own function (maybe `calculate_http_endpoint`)